src/services/StructuralMetadataUtils.js

```diff
@@ -68,45 +68,50 @@ export default class StructuralMetadataUtils {
    * @param {Float} duration - end time of the media file in Milliseconds
    */
   buildSMUI(allItems, duration) {
-    // Regex to match mm:ss OR single number
-    const regexMMSS = /^([0-9]*:[0-9]*.[0-9]*)$/i;
+    // Regex to match mm:ss OR seconds(as a float)/minutes(as an int)
+    const regexHHMMSS = /^([0-9]*:[0-9]*:[0-9]*.[0-9]*)$/i;
     const regexSS = /^([0-9]*.[0-9]*)$/i;
 
+    // Convert file duration to seconds
+    const durationInSeconds = Math.round(duration / 10) / 100;
+
     // Convert time to HH:mm:ss.ms format to use in validation logic
-    let convertToHHmmss = time => {
-      if (regexMMSS.test(time)) {
-        let [minutes, seconds] = time.split(':');
-        let minutesIn = parseInt(minutes) * 60;
-        let secondsIn = seconds === '' ? 0.0 : parseFloat(seconds);
-        return minutesIn + secondsIn;
-      }
+    let convertToSeconds = time => {
       if (regexSS.test(time)) {
-        return parseFloat(time);
-      } else {
-        return this.toMs(time) / 1000;
+        let [minutes, seconds] = time.split('.');
+        let minutesInS = parseInt(minutes) * 60;
+        let secondsNum = seconds ? parseFloat(seconds) : 0.0;
+        return minutesInS + secondsNum;
       }
+      if (regexHHMMSS.test(time)) {
+        let [seconds, minutes, hours] = time.split(':').reverse();
+        let hoursInS = hours ? parseInt(hours) * 3600 : 0;
+        let minutesInS = parseInt(minutes) * 60;
+        let secondsNum = seconds === '' ? 0.0 : parseFloat(seconds);
+        return hoursInS + minutesInS + secondsNum;
+      }
+      return this.toMs(time) / 1000;
     };
 
-    let stripHTMLCodes = lableText => {
-      return lableText
-                .replace(/&amp;/g, "&")
-                .replace(/&lt;/g, "<")
-                .replace(/&gt;/g, ">")
-                .replace(/&quot;/g, '"')
-                .replace(/&apos;/g, "'")
-    }
+    let decodeHTML = lableText => {
+      var txt = document.createElement('textarea');
+      txt.innerHTML = lableText;
+      return txt.value;
+    };
```

Comment:
@adamjarling I found this as one of the popular ways to strip HTML entities from strings in JavaScript. What do you think? :)

Reply:
Hmm, interesting workaround that I'm sure would do the trick.  It does introduce an extra element onto the DOM, which hopefully wouldn't mess things up?   Honestly I'd prefer your original way of doing it if there are no other JS methods which handle encoding conversion...seems like we might scratch our heads in a year or two, for why we added a textarea?

Reply:
Yeah, this looks a bit messy.. I couldn't find the textarea in the DOM when application is loaded. Maybe I'm looking at the wrong place? :) Let me see whether I can use regular expressions to solve this instead of matching each and every character.

Reply:
I couldn't find a better way with regular expressions. Should I revert this to the original function?

Reply:
@adamjarling I couldn't find a better way with regular expressions. Should I revert this to the original function?

Reply:
Yea, sorry.  Let's revert it maybe?

Reply:
No worries. While writing that, even I thought that function needs to be more robust. Okay, let's keep it until we come across a better solution in the future :+1: